src/lib/y2storage/guided_proposal.rb

```diff
@@ -57,16 +57,81 @@ class GuidedProposal
+      #
+      # If a proposal is not possible by honoring current settings, other settings
+      # are tried. For example, a proposal without separate home or without snapshots
+      # will be calculated.
+      #
+      # @see GuidedProposal#initialize
+      #
+      # @param settings [ProposalSettings] if nil, default settings will be used
+      # @param devicegraph [Devicegraph] starting point. If nil, the probed
+      #   devicegraph will be used
+      # @param disk_analyzer [DiskAnalyzer] if nil, a new one will be created
+      #   based on the initial devicegraph.
+      #
+      # @return [GuidedProposal]
+      def initial(settings: nil, devicegraph: nil, disk_analyzer: nil)
+        # Try proposal with initial settings
+        current_settings = settings || ProposalSettings.new_for_current_product
+        log.info("Trying proposal with initial settings: #{current_settings}")
+        proposal = try_proposal(current_settings.dup, devicegraph, disk_analyzer)
+
+        # Try proposal without home
+        if proposal.failed? && current_settings.use_separate_home
+          current_settings.use_separate_home = false
+          log.info("Trying proposal without home: #{current_settings}")
+          proposal = try_proposal(current_settings.dup, devicegraph, disk_analyzer)
+        end
+
+        # Try proposal without snapshots
+        if proposal.failed? && current_settings.snapshots_active?
+          current_settings.use_snapshots = false
+          log.info("Trying proposal without home neither snapshots: #{current_settings}")
+          proposal = try_proposal(current_settings.dup, devicegraph, disk_analyzer)
+        end
+
+        proposal
+      end
+
+    private
+
+      # Try a proposal with specific settings. Always returns the proposal, even
+      # when it is not possible to make a valid one. In that case, the resulting
+      # proposal will not have devices.
+      #
+      # @return [GuidedProposal]
+      def try_proposal(settings, devicegraph, disk_analyzer)
+        proposal = GuidedProposal.new(
+          settings:      settings,
+          devicegraph:   devicegraph,
+          disk_analyzer: disk_analyzer
+        )
+        proposal.propose
+        proposal
+      rescue Y2Storage::Error => e
+        log.error("Proposal failed: #{e.inspect}")
+        proposal
+      end
+    end
+
     # @param settings [ProposalSettings] if nil, default settings will be used
     # @param devicegraph [Devicegraph] starting point. If nil, the probed
     #   devicegraph will be used
     # @param disk_analyzer [DiskAnalyzer] if nil, a new one will be created
-    #   based in the initial devicegraph
+    #   based on the initial devicegraph
     def initialize(settings: nil, devicegraph: nil, disk_analyzer: nil)
       @settings = settings || ProposalSettings.new_for_current_product
       @proposed = false
-      @initial_graph = devicegraph
-      @disk_analyzer = disk_analyzer
+      if devicegraph.nil?
```

Comment:
I like this change. But not `#initial_graph` should become a straightforward `attr_reader`. Otherwise it's confusing.